core/vtk/ttkFiberSurface/ttkFiberSurface.cpp

```diff
@@ -114,15 +114,21 @@ int ttkFiberSurface::RequestData(vtkInformation *ttkNotUsed(request),
 
   inputPolygon_.clear();
 
+  SimplexId cellNumber = polygon->GetNumberOfCells();
+  vtkCellArray *connectivity = polygon->GetCells();
+
+  if(connectivity->GetData()->GetNumberOfTuples() < 3 * cellNumber) {
+    this->printErr("Error: ill-defined range polygon.");
+    return 0;
+  }
+
 #if !defined(_WIN32) || defined(_WIN32) && defined(VTK_USE_64BIT_IDS)
-  const long long int *cellArray
-    = polygon->GetCells()->GetData()->GetPointer(0);
+  const long long int *cellArray = connectivity->GetData()->GetPointer(0);
 #else
-  int *pt = polygon->GetCells()->GetPointer();
+  int *pt = connectivity->GetPointer();
```

Comment:
`vtkIdType` would be safer here ? (@julien-tierny is `SimplexId` defined as `vtkIdType` ?)